app/connectors/HipConnector.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import uk.gov.hmrc.http.{HeaderCarrier, HeaderNames}
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+
+import java.util.UUID
+
+trait HipConnector {
+
+  val config: ServicesConfig
+
+  lazy val hipBaseUrl: String = config.baseUrl("hip")
+  lazy val hipAuthToken: String = config.getString("microservice.services.hip.authorisation-token")
```

Comment:
Should be eagerly evaluated and not lazily so that it we will get errors for the missing config on application start.

Reply:
Good spot, will have to move the mixin to the Class instead of the trait though. Otherwise the mock[] instance of the interface blows up.

Reply:
Changed on [fc16d24](https://github.com/hmrc/income-tax-interest/pull/68/commits/fc16d2406f68463375f7c863c71786a4288b82fe)